backend/app/admin.py

```diff
@@ -0,0 +1,95 @@
+"""
+Django admin configuration for Todo application.
+
+This module defines the admin interface for managing Todo and Task instances.
+"""
+
+from django.contrib import admin
+from .models import Todo, Task
+
+
+class TodoInline(admin.TabularInline):
+    """Inline admin for managing todos within a task."""
+    model = Todo
+    fk_name = 'task'
+    extra = 1
+    verbose_name = "Todo"
+    verbose_name_plural = "Todos"
+    fields = ['title', 'description', 'completed', 'due_date']
+
+
+@admin.register(Todo)
+class TodoAdmin(admin.ModelAdmin):
+    """Admin interface for Todo model."""
+    list_display = (
+        'id', 'title', 'completed', 'due_date',
+        'created_at', 'updated_at', 'tags', 'description'
+    )
+    list_filter = (
+        'completed', 'tags', 'task', 'user', 'created_at', 'updated_at'
+    )
+    search_fields = ('title', 'description', 'tags')
+    list_editable = ('completed',)
+    list_per_page = 25
+    date_hierarchy = 'created_at'
+    fieldsets = (
+        ('Basic Information', {
+            'fields': ('title', 'description', 'completed')
+        }),
+        ('Timing', {
+            'fields': ('due_date', 'created_at', 'updated_at'),
+            'classes': ('collapse',)
+        }),
+        ('Organization', {
+            'fields': ('tags', 'user', 'task')
+        }),
+    )
+    readonly_fields = ('created_at', 'updated_at')
+
+
+@admin.register(Task)
+class TaskAdmin(admin.ModelAdmin):
+    """Admin interface for Task model with inline todo editing."""
+    list_display = (
+        'id', 'title', 'user', 'created_at', 'updated_at', 'todos_count'
+    )
+    list_filter = ('user', 'created_at', 'updated_at')
+    search_fields = ('title', 'description', 'user__username')
+    list_per_page = 25
+    date_hierarchy = 'created_at'
+    fieldsets = (
+        ('Basic Information', {
+            'fields': ('title', 'description', 'user')
+        }),
+        ('Timing', {
+            'fields': ('created_at', 'updated_at'),
+            'classes': ('collapse',)
+        }),
+    )
+    readonly_fields = ('created_at', 'updated_at')
+    inlines = [TodoInline]
+
+    def todos_count(self, obj):
+        """Return the number of todos in this task."""
+        return obj.todos.count()
```

Comment:
this is going to be slow, fyi

it's called N+1 query problem, perhaps not a good fit for proudction code

consider either removing this feature or use the query set itself to aggregate the count in the `get_queryset`